pkg/apis/rke.cattle.io/v1/etcdsnapshot_types.go

```diff
@@ -2,28 +2,38 @@ package v1
 
 import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 
+// ETCDSnapshotS3 defines S3 snapshot configuration for ETCD backups.
 type ETCDSnapshotS3 struct {
 	// Endpoint is the S3 endpoint used for snapshot operations.
+	// If this field is not explicitly set, the 'defaultEndpoint' value from the referenced CloudCredential will be used.
 	// +nullable
 	// +optional
 	Endpoint string `json:"endpoint,omitempty"`
 
 	// EndpointCA is the CA certificate for validating the S3 endpoint.
+	// This can be either a file path (e.g., "/etc/ssl/certs/my-ca.crt")
+	// or the CA certificate content, in base64-encoded or plain PEM format.
+	// If this field is not explicitly set, the 'defaultEndpointCA' value from the referenced CloudCredential will be used.
 	// +nullable
 	// +optional
 	EndpointCA string `json:"endpointCA,omitempty"`
 
-	// SkipSSLVerify is a flag used to ignore certificate validation errors
-	// when using the configured endpoint.
+	// SkipSSLVerify defines whether TLS certificate verification is disabled.
+	// If this field is not explicitly set, the 'defaultSkipSSLVerify' value
+	// from the referenced CloudCredential will be used.
 	// +optional
 	SkipSSLVerify bool `json:"skipSSLVerify,omitempty"`
 
 	// Bucket is the name of the S3 bucket used for snapshot operations.
+	// If this field is not explicitly set, the 'defaultBucket' value from the referenced CloudCredential will be used.
+	// An empty bucket name will cause a 'failed to initialize S3 client: s3 bucket name was not set' error.
+	// +kubebuilder:validation:MaxLength=63
```

Comment:
Didn't introduce MinLength or marked it as required because the value could come from cloud credential.